hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientSmartInvocationServiceImpl.java

```diff
@@ -66,16 +65,16 @@ public void invokeOnTarget(ClientInvocation invocation, Address target) throws I
         invokeOnConnection(invocation, (ClientConnection) connection);
     }
 
-    private Connection getOrTriggerConnect(Address target) throws IOException {
+    private Connection getOrTriggerConnect(Address target) {
         Connection connection = connectionManager.getOrTriggerConnect(target);
         if (connection == null) {
-            throw new IOException("No available connection to address " + target);
+            throw new RetryableHazelcastException("No available connection to address " + target);
```

Comment:
If such an exception is going to end up in the face of the end user, I would expect a concrete subclass of this exception. For and end user it is very confusing to get a stacktrace for something that was says it will automatically be retried.